### PR TITLE
Update wrapper and rename SourceFileHeader constants

### LINKED ISSUE #29715: Replace constants in SourceFileHeader
### Issue type

Polishing

### Problem description

Once https://github.com/gradle/gradle/pull/29638 is merged in, I should update the wrapper and replace the old constants in `SourceFileHeader` with the new, properly named ones (which should be in the new wrapper).

### Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by jbartok**
Fixes #29715 

### Contributor Checklist
- [ ] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md).
- [ ] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [ ] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [ ] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team.
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective.
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic.
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes.
- [ ] Ensure that tests pass sanity check: `./gradlew sanityCheck`.
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by jbartok** [comment]
@bot-gradle APT

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.9-rc-1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions-snapshots/gradle-8.10-20240701001954+0000-bin.zip
```
Comment (ov7a) [inline gradle/wrapper/gradle-wrapper.properties:3]: ❓ depending on the snapshot wrapper bothers me. This means our builds are harder to reproduce. I don't feel pleasuring the formatted worth it. I'd delay this change

Comment (jbartok) [reply gradle/wrapper/gradle-wrapper.properties:3]: Makes sense to me, I can leave this PR open until there is a new, non-snapshot distribution available.

Comment (eskatos) [reply gradle/wrapper/gradle-wrapper.properties:3]: You can trigger a nightly promotion yourself in these cases if you don't wanna wait.

Comment (jbartok) [reply gradle/wrapper/gradle-wrapper.properties:3]: That would still be a snapshot, wouldn't it? Well, "snapshot distribution"... But now that I give it more thought, these aren't really snapshots, are they? None of them will ever be changed or overwritten, so I'm not sure that the initial concern about the builds being harder to reproduce is valid. Wdyt?

Comment (ov7a) [reply gradle/wrapper/gradle-wrapper.properties:3]: > None of them will ever be changed or overwritten,

They are deleted after some time, IRC

**Comment by jbartok** [comment]
@bot-gradle test this

**Pull request merged**